src/main/java/uk/gov/hmcts/sptribs/caseworker/event/CaseWorkerIssueDecision.java

```diff
@@ -36,11 +42,22 @@ public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuild
             .name("Issue a decision")
             .description("Issue a decision")
             .showEventNotes()
+            .aboutToSubmitCallback(this::aboutToSubmit)
             .grant(CREATE_READ_UPDATE_DELETE, COURT_ADMIN_CIC, SUPER_USER)
             .grantHistoryOnly(SOLICITOR));
         issueDecisionNotice.addTo(pageBuilder);
         issueDecisionSelectTemplate.addTo(pageBuilder);
         issueDecisionUploadNotice.addTo(pageBuilder);
+        issueDecisionSelectRecipients.addTo(pageBuilder);
+    }
+
+    public AboutToStartOrSubmitResponse<CaseData, State> aboutToSubmit(CaseDetails<CaseData, State> details,
+                                                                       CaseDetails<CaseData, State> beforeDetails) {
+        var caseData = details.getData();
+        return AboutToStartOrSubmitResponse.<CaseData, State>builder()
+            .data(caseData)
+            .state(CaseManagement)//or AwaitingHearing
```

Comment:
In State Transition diagram, think we need to move to CaseManagement/Awaitinghearing based on a condition. Is this logic covered as part of some other jiira?

Reply:
As per Jeff Nayler's confirmation, state is not implemented for this particular JIRA

Reply:
This will be covered as part of another jiira.